axum/src/handler/mod.rs

```diff
@@ -284,15 +283,13 @@ where
     }
 }
 
-impl<S, T, ReqBody, ResBody> Handler<T, ReqBody> for Layered<S, T>
+impl<S, T, ReqBody> Handler<T, ReqBody> for Layered<S, T>
 where
-    S: Service<Request<ReqBody>, Response = Response<ResBody>> + Clone + Send + 'static,
-    S::Error: IntoResponse,
+    S: Service<Request<ReqBody>, Error = Infallible> + Clone + Send + 'static,
+    S::Response: IntoResponse,
```

Comment:
Think I'll just change this here as well since its touching the same code. Handler's wrapped in middleware should be infallible just like everything else.